src/main/java/frc/robot/subsystems/leds/LEDSubsystem.java

```diff
@@ -38,20 +50,116 @@ public void periodic() {
         // Toggle the blink boolean every duration to swap the LEDs between the driver piece color
         // and the blink color.
         if (blinkTimer.advanceIfElapsed(BLINK_DURATION_SECONDS)) blinking = !blinking;
-        ledStrip.setSolidColor(blinking ? BLINK_COLOR : color);
+        if(manual){
+            //if manual and continuous set the bottom to color
+            ledStrip.updateContinuousColor(color);
+            ledStrip.setContinuousColor();
+        //} else if(manual){
+            //if manual and not continuous fill entire strip with color
+            // ledStrip.fillContinuousColor(color);
+            // ledStrip.setContinuousColor();
+        } else {
+            setColorPulse();
+            //setTwoColor()
+        }
     }
 
     /**
-     * Sets the driver-provided color of the LEDs.
-     * @param red The red component of the color.
-     * @param green The green component of the color.
-     * @param blue The blue component of the color.
+     * Sets the driver-provided color of the LEDs from provided RGB values.
+     * @param r The red component of the color, from [0, 255].
+     * @param g The green component of the color, from [0, 255].
+     * @param b The blue component of the color, from [0, 255].
      */
-    public void setColor(double red, double green, double blue) {
+    public void setRGB(double r, double g, double b) {
+        //if the color doesnt change, dont do anything
+        if(this.color.red == r && this.color.green == g && this.color.blue == b){ 
+            return;
+        }
         this.color = new Color(
-            (int) (red * BRIGHTNESS_SCALE_FACTOR),
-            (int) (green * BRIGHTNESS_SCALE_FACTOR),
-            (int) (blue * BRIGHTNESS_SCALE_FACTOR)
+            (int) (r * BRIGHTNESS_SCALE_FACTOR),
+            (int) (g * BRIGHTNESS_SCALE_FACTOR),
+            (int) (b * BRIGHTNESS_SCALE_FACTOR)
+        );
+        //sets the whole strip to the color
+        ledStrip.fillContinuousColor(this.color);
+        ledStrip.setContinuousColor();
+    }
+
+    /**
+     * Sets the driver-provided color of the LEDs from provided HSV values.
+     * @param h The hue component of the color, from [0, 180).
+     * @param s The saturation component of the color, from [0, 255].
+     * @param v The value component of the color, from [0, 255].
+     */
+    //sets the color to the 
+    public void setHSV(double h, double s, double v) {
+        this.color = Color.fromHSV(
+            (int) h,
+            (int) s,
+            (int) (v * BRIGHTNESS_SCALE_FACTOR)
         );
     }
+
+    public void toggleLEDControlMode(){
+        continuous = !continuous;
+    }
+
+    public void setManual(boolean manual){
+        this.manual = manual;
+    }
+
+    public void tagDetected(){
```

Comment:
Is there any documentation on what `aprilTimer` and `aprilTimer2` control? From what I can gather `aprilTimer` is used to send a color pulse and `aprilTimer2` controls the length of a 2-color display(?), but when is color pulse used and when is the two color system used? Are they ever used together?

Reply:
april timer and april timer 2 both are counting the time since an april tag was last seen